Add OpenAPI spec, generated client and sanitation (#2)
ballerina-platform/module-ballerinax-openai.finetunes: docs/spec/sanitations.md

_Authors_: @kanishkagu  \
_Created_: 2024/08/05 \
_Updated_: 2024/08/05 \
_Edition_: Swan Lake

# Sanitation for OpenAPI specification

This document records the sanitation done on top of the official OpenAPI specification from OpenAI Finetune. The OpenAPI specification is obtained from the [OpenAPI specification for the OpenAI API](https://github.com/openai/openai-openapi/blob/master/openapi.yaml). These changes are implemented to enhance the overall usability and readability of the generated client.

> 1. **Change the status_details property parameters of the 'OpenAIFile' object**:

Reviewer: ```suggestion
1. **Changed the `status_details` property parameter of the `OpenAIFile` object**:
```

   - **Original**:
      - Deprecated: `true`
      - Nullable parameter: Not included

   - **Updated**:
      - Removed the `deprecated` parameter
      - Added the `nullable` parameter as `true`

   - **Reasons**: The original configuration was generated successfully, but it caused a compile-time error. Updating the parameters resolved this error, enhancing the overall functionality and ensuring smooth compilation.

> 2. **Removed the `default:null` property of certain schemas**:

Reviewer: ```suggestion
2. **Removed the `default:null` property from the following schemas**:
```

   - **Changed Schemas**: `CreateCompletionRequest`,`ChatCompletionStreamOptions`,`CreateChatCompletionRequest`

>    - **Original**:
>       - defaullt: `null`

Reviewer: ```suggestion
   - **Original**:
      - default: `null`
```

   - **Updated**:
      - Removed the `default` parameter 

   - **Reason**: This change is done as a workaround for ballerina openapi tool not allowing to generate the client.

> 3. **Removing the Required Field from the Error Property of FineTuningJob Object**

Reviewer: The convention is to use past tense for the sanitation items. 
```suggestion
3. **Removed the Required Field from the `Error` Property of `FineTuningJob` Object`**
```
